File: 프로그래머스/0/181927. 마지막 두 원소/마지막 두 원소.java
import java.util.*;

class Solution {
    public int[] solution(int[] num_list) {
        int len = num_list.length;
        int[] tmpArr = Arrays.copyOf(num_list, len);
        int[] answer = new int[len+1];
        
        int max = tmpArr[len-1];
        int subMax = tmpArr[len-2];
        
        for(int i=0; i<len; i++){
            answer[i] = tmpArr[i];
        }
        if(max > subMax) answer[answer.length-1] = max - subMax;
        else answer[answer.length-1] = max *=2;
        
        return answer;
    }
}

File: 프로그래머스/0/181927. 마지막 두 원소/마지막 두 원소.kt
class Solution {
    fun solution(num_list: IntArray): IntArray{
        val answer: MutableList<Int> = num_list.copyOf().toMutableList()
                        
        if(num_list.last() > num_list[num_list.size - 2] ){
                answer.add(num_list.last() - num_list[num_list.size - 2]  )
        }else{
                answer.add(num_list.last() * 2)
        }

        return answer.toIntArray()
    }
}